lib/src/editor.dart

```diff
@@ -0,0 +1,790 @@
+import 'package:yaml/src/equality.dart';
+import 'package:yaml/yaml.dart';
+
+import './source_edit.dart';
+
+/// An interface for modififying [YAML][1] documents while preserving comments and
+/// whitespaces.
+///
+/// YAML parsing is supported by `package:yaml`, and modifications are performed as
+/// string operations. Each time a modification takes place via one of the public
+/// methods, we calculate the expected final result, and parse the result YAML string,
+/// and ensure the two YAML trees match. Users may define the default settings to be
```

Comment:
Describe what happens if they don't match, and when such a thing can be expected (if ever)

Reply:
Fixed, thanks! :)